forest/oak/base.py

```diff
@@ -578,21 +578,39 @@ def run(study_folder: str, output_folder: str, tz_str: Optional[str] = None,
         )
 
         days = pd.date_range(date_start, date_end, freq='D')
-        if (frequency == Frequency.HOURLY_AND_DAILY
-                or frequency == Frequency.HOURLY):
-            freq = 'H'
-        else:
-            freq = str(frequency.value) + 'H'
-        days_hourly = pd.date_range(date_start, date_end+timedelta(days=1),
-                                    freq=freq)[:-1]
+
         # allocate memory
         steps_daily = np.full((len(days), 1), np.nan)
         cadence_daily = np.full((len(days), 1), np.nan)
         walkingtime_daily = np.full((len(days), 1), np.nan)
 
-        steps_hourly = np.full((len(days_hourly), 1), np.nan)
-        cadence_hourly = np.full((len(days_hourly), 1), np.nan)
-        walkingtime_hourly = np.full((len(days_hourly), 1), np.nan)
+        if frequency != Frequency.DAILY:
+            if (
+                frequency == Frequency.HOURLY_AND_DAILY
+                or frequency == Frequency.HOURLY
+            ):
+                freq = 'H'
+            elif frequency == Frequency.MINUTELY:
+                freq = 'T'
+            else:
+                freq = str(frequency.value) + 'H'
+
+            days_hourly = pd.date_range(date_start, date_end+timedelta(days=1),
+                                        freq=freq)[:-1]
+
+            steps_hourly = np.full((len(days_hourly), 1), np.nan)
+            cadence_hourly = np.full((len(days_hourly), 1), np.nan)
+            walkingtime_hourly = np.full((len(days_hourly), 1), np.nan)
+
+            t_ind_pydate = days_hourly.to_pydatetime()
+            t_ind_pydate_str = t_ind_pydate.astype(str)
+        else:
+            # set them to None to avoid reference before assignment
+            steps_hourly = np.full((1, 1), np.nan)
+            cadence_hourly = np.full((1, 1), np.nan)
+            walkingtime_hourly = np.full((1, 1), np.nan)
+            t_ind_pydate = None
+            t_ind_pydate_str = None
```

Comment:
Would it be more readable to move these initializations to the others above on lines 583-585?